### DISCUSSION TITLE
Error while generating a database for V8

### DISCUSSION THREAD

**Discussion by h3ku** [github/securitylab#112] [General] [answered]
Hey,

Not sure if this is the best place to post this, in any case I think this could be useful for others in the future.

Im trying to create a database for V8, but so far I didn't manage to make it work.

Im using the following command to build V8
`codeql database create v8-database --language=cpp --command="ninja -C out/ d8"`

It builds fine, but the database generation process fails with the following error `A fatal error occurred: No source code was seen during the build`

From what I see Ninja is supported "https://help.semmle.com/wiki/pages/viewpage.action?pageId=24085367", so no idea of why this is failing.

You can find the full log here: https://gist.github.com/h3ku/f426e6f302e0ff4a6046a099103bd2e6

Pd: I know that there is already a pregenerated database for V8 (https://semmle.com/large-oss-projects), but its quite old.

**Comment by kevinbackhouse** [comment]
I am not familiar with the V8 build, but there are a few common reasons for
that error message. The first possibility is that the build is happening in
a completely different directory than the one that you ran the command in.
CodeQL is only monitoring the current directory, so it won't see source
files that are built elsewhere. The second possibility is that the build is
using caching. You need to make sure that it is doing a full rebuild so
that CodeQL sees all the compiler invocations.

Hope that helps,

Kev

**Comment by h3ku** [comment, marked as answer]
Thanks Kevin!
@matt-gretton-dann pointed out in slack that there were some timeout errors in the output log and suggested to run ninja single-threaded with -j1 'codeql database create ... --command="ninja -j1 -C out/ d8"'

It takes a while to compile as expected, but it works, codeql successfully generated the DB this way!
